apps/exam/views.py

```diff
@@ -59,7 +62,14 @@ def post(self, request, exam_id):
 
     current_attempt = ExamAttempt.objects.create(
         user=request.user, exam=exam, duration_minutes=exam_duration_minutes)
-    for question in exam.question_set.all():
+
+    if exam.question_set.all().count() < exam.question_count:
+      question_pool_ids = [question.id for question in exam.question_set.all()]
+    else:
+      question_pool_ids = random.sample(range(exam.question_set.all().count()),
```

Comment:
I'd use a sample based on all exam question ids. The current sample would always work as expected.